File: javascript/protein-translation/protein-translation.js
export const translate = (codons) => {
	if (!codons) { return []; }

	let protein = [];

	let parsedCodons = codons.match(/.{1,3}/g);

	for (let i = 0; i < parsedCodons.length; i++) {
		switch (parsedCodons[i]) {
			case 'AUG':
				protein.push('Methionine');
				break;
			case 'UUU':
			case 'UUC':
				protein.push('Phenylalanine');
				break;
			case 'UUA':
			case 'UUG':
				protein.push('Leucine');
				break;
			case 'UCU':
			case 'UCC':
			case 'UCA':
			case 'UCG':
				protein.push('Serine');
				break;
			case 'UAU':
			case 'UAC':
				protein.push('Tyrosine');
				break;
			case 'UGU':
			case 'UGC':
				protein.push('Cysteine');
				break;
			case 'UGG':
				protein.push('Tryptophan');
				break;
			case 'UAA':
			case 'UAG':
			case 'UGA':
				return protein
			default:
				throw 'Invalid codon';
		}		
	}
	
	return protein;
};

File: csharp/protein-translation/ProteinTranslation.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ProteinTranslation
{
    public static string[] Proteins(string strand)
    {
        const string StopSequence = "STOP";
        const int CodonSize = 3;

        Dictionary<string, string> proteinMapping = new Dictionary<string, string>()
        {
            { "AUG", "Methionine" },
            { "UUU", "Phenylalanine"},
            { "UUC", "Phenylalanine"},
            { "UUA", "Leucine"},
            { "UUG", "Leucine"},
            { "UCU", "Serine"},
            { "UCC", "Serine"},
            { "UCA", "Serine"},
            { "UCG", "Serine"},
            { "UAU", "Tyrosine"},
            { "UAC", "Tyrosine"},
            { "UGU", "Cysteine"},
            { "UGC", "Cysteine"},
            { "UGG", "Tryptophan"},
            { "UAA", StopSequence},
            { "UAG", StopSequence},
            { "UGA", StopSequence},
        };

        var codons = Enumerable.Range(0, (strand.Length + CodonSize - 1) / CodonSize)
                               .Select(i => proteinMapping[strand.Substring(i * CodonSize, CodonSize)])
                               .ToArray();
        var stopSequenceIndex = Array.IndexOf(codons, StopSequence);

        return stopSequenceIndex != -1 ? Enumerable.Range(0, stopSequenceIndex).Select(i => codons[i]).ToArray() : codons;
            //Array.Empty<string>() : codons;
    }
}